src/test/regress/expected/subselect_optimizer.out

```diff
@@ -2245,3 +2245,36 @@ drop table tl1;
 drop table tl2;
 drop table tl3;
 drop table tl4;
+--Test case for subquery, which returns more than one rows
+create table table1 as
+    select * from (values (1, 0), (1, 0)) v(a, b) distributed by (a);
+create table table2 as
+    select * from (values (0, 10), (0, 10)) v(a, b) distributed by (a);
+explain (costs off)
+select * from table1 where 10 in
+    (select b from table2 where table2.a = 0 or table1.b = table2.b);
+                          QUERY PLAN                          
+--------------------------------------------------------------
+ Result
+   Filter: (SubPlan 1)
+   ->  Gather Motion 3:1  (slice1; segments: 3)
+         ->  Seq Scan on table1
+   SubPlan 1
+     ->  Result
+           Filter: ((table2.a = 0) OR (table1.b = table2.b))
+           ->  Materialize
+                 ->  Gather Motion 3:1  (slice2; segments: 3)
+                       ->  Seq Scan on table2
+                             Filter: (10 = b)
+ Optimizer: GPORCA
+(12 rows)
+
+select * from table1 where 10 in
+    (select b from table2 where table2.a = 0 or table1.b = table2.b);
```

Comment:
The query can be rewritten as
```
select * from table1 where exists
    (select from table2 where (b = 10) and (table2.a = 0 or table1.b = table2.b));
```

This query can be executed successfully before the patch.

Its plan is the same
```
 Result
   Filter: (SubPlan 1)
   ->  Gather Motion 3:1  (slice1; segments: 3)
         ->  Seq Scan on table1
   SubPlan 1
     ->  Result
           Filter: ((table2.a = 0) OR (table1.b = table2.b))
           ->  Materialize
                 ->  Gather Motion 3:1  (slice2; segments: 3)
                       ->  Seq Scan on table2
                             Filter: (b = 10)
 Optimizer: GPORCA
(12 rows)
```

What about adding transformation to get plan with EXISTS_SUBLINK?

Reply:
Plan of the first query before patch contained one more Result node at the top of the subplan. It was because Orca generated incorrect type of subplan: scalar. I think, that it does not matter EXISTS_SUBLINK or ANY_SUBLINK. And ANY_SUBLINK is more correct for the first query, because there is comparison between int and array.